storage/country_tree.hpp

```diff
@@ -157,18 +147,34 @@ class CountryTree
   using TCountryTreeHashTable = unordered_multimap<TKey, Node *>;
 
 public:
-  CountryTree(TValue const & value = TValue(), Node * parent = nullptr)
-    : m_countryTree(make_unique<Node>(value, parent))
+  bool IsEmpty() const { return m_countryTree == nullptr; }
+
+  Node const & GetRoot() const
   {
+    CHECK(m_countryTree, ());
+    return *m_countryTree;
   }
 
-  Node const & GetRoot() const { return *m_countryTree; }
-  Node & GetRoot() { return *m_countryTree; }
+  Node & GetRoot()
+  {
+    CHECK(m_countryTree, ());
+    return *m_countryTree;
+  }
 
   /// @return reference is valid only up to the next tree structure modification
   TValue & AddAtDepth(int level, TValue const & value)
   {
-    Node * const added = m_countryTree->AddAtDepth(level, value);
+    Node * added = nullptr;
```

Comment:
It does not work this way. 
1. I'd like to have an assert if somebody calls AddAtDepth with zero level on not empty country tree
2. Now Node::AddAtDepth() is not ready for calling with level zero. There's an assert inside Node::AddAtDepth()

As you see, the numeration of level is changed now (in this PR). Now zero means the root. The first level is countries.